tests/test_docker_validator.py

```diff
@@ -113,39 +122,87 @@ def test_validate(self):
 
-        with open(self.validator.metadata_json_post_validation) as open_file:
+        with open(validator.metadata_json_post_validation) as open_file:
             json_data = json.load(open_file)
-            assert json_data.get('files') == [
-                {'analysisAlias': 'AA', 'fileName': 'input_passed.vcf', 'fileType': 'vcf',
-                 'md5': '96a80c9368cc3c37095c86fbe6044fb2', 'fileSize': 45050}
-            ]
+            assert json_data.get('files') == expected_metadata_files_json
 
         # Check metadata errors
-        with open(os.path.join(self.validator.output_dir, 'other_validations', 'metadata_validation.txt')) as open_file:
+        with open(os.path.join(validator.output_dir, 'other_validations', 'metadata_validation.txt')) as open_file:
             metadata_val_lines = {l.strip() for l in open_file.readlines()}
-            assert 'must match pattern "^PRJ(EB|NA)\\d+$"' in metadata_val_lines
+            assert any((expected_metadata_val in line for line in metadata_val_lines))
 
         # Check semantic metadata errors
-        semantic_yaml_file = os.path.join(self.validator.output_dir, 'other_validations', 'metadata_semantic_check.yml')
+        semantic_yaml_file = os.path.join(validator.output_dir, 'other_validations', 'metadata_semantic_check.yml')
         self.assertTrue(os.path.isfile(semantic_yaml_file))
         with open(semantic_yaml_file) as open_yaml:
             semantic_output = yaml.safe_load(open_yaml)
-            assert semantic_output[1] == {'description': 'SAME123 does not exist or is private',
+            assert semantic_output[1] == expected_semantic_val
+
+    def test_validate(self):
+        # run validation in docker
+        self.validator.validate()
+        expected_sample_checker = {
+            'overall_differences': False,
+            'results_per_analysis': {
+                'AA': {
+                    'difference': False,
+                    'more_metadata_submitted_files': [],
+                    'more_per_submitted_files_metadata': {},
+                    'more_submitted_files_metadata': []
+                }
+            }
+        }
+        expected_metadata_files_json = [
+            {'analysisAlias': 'AA', 'fileName': 'input_passed.vcf', 'fileType': 'vcf',
+             'md5': '96a80c9368cc3c37095c86fbe6044fb2', 'fileSize': 45050}
+        ]
+        expected_metadata_val = 'must match pattern "^PRJ(EB|NA)\\d+$"'
+        expected_semantic_val = {'description': 'SAME123 does not exist or is private',
                                           'property': '/sample/0/bioSampleAccession'}
 
+        self.assess_post_validation(self.validator, expected_sample_checker, expected_metadata_files_json,
+                                    expected_metadata_val, expected_semantic_val)
+
     def test_validate_from_excel(self):
         self.validator_from_excel.validate()
         self.assertTrue(os.path.isfile(self.validator_from_excel._sample_check_yaml))
```

Comment:
```suggestion
```
If I understand the refactoring correctly, this is covered by `assert_sample_checker`